x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/search_bar/index.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiFieldSearch, EuiButton } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+export interface SearchBarProps {
+  defaultValue?: string;
+  onSearch(value: string): void;
+}
+
+export const SearchBar = memo<SearchBarProps>(({ defaultValue = '', onSearch }) => {
+  const [query, setQuery] = useState<string>(defaultValue);
+  return (
+    <EuiFlexGroup direction="row" alignItems="center" gutterSize="l">
+      <EuiFlexItem>
+        <EuiFieldSearch
+          defaultValue={query}
+          placeholder={i18n.translate(
+            'xpack.securitySolution.trustedapps.list.search.placeholder',
+            {
+              defaultMessage: 'Search',
+            }
+          )}
+          onChange={(e) => setQuery(e.target.value)}
```

Comment:
consider creating this callback handler with `useCallback()`. Same comment for the callbck below on line 39.